### DISCUSSION TITLE
Clarification on NATS messages published to MQTT subscriptions are always delivered as QoS 0 messages

### DISCUSSION THREAD

**Discussion by wilianto** [nats-io/nats-server#4750] [Q&A] [answered]
I'm seeking clarification regarding the current limitations of NATS MQTT. On the official [page](https://docs.nats.io/running-a-nats-service/configuration/mqtt), there appears to be conflicting information. Initially, it states:

> If the MQTT subscription is QoS1 and an MQTT publisher publishes an MQTT QoS1 message on the same or any other server in the cluster, the message will be persisted in the cluster and routed and delivered as QoS 1 to the MQTT subscription.

> When the server delivers a QoS 1 message to a QoS 1 subscription, it will keep the message until it receives the PUBACK for the corresponding packet identifier. If it does not receive it within the "ack_wait" interval, that message will be resent.

However, in the subsequent limitation section, it is mentioned:

> NATS messages published to MQTT subscriptions are always delivered as QoS 0 messages.

I'm seeking assistance in understanding which of these statements is accurate and reflective of the current behavior.

**Comment by levb** [comment, marked as answer]
@wilianto in addition to messages that are submitted by other MQTT clients, an MQTT subscriber can receive other NATS messages, including those submitted by other "pure NATS" clients. Those "pure NATS" are always delivered to an MQTT subscriber as QOS 0 (because we wouldn't know how else to interpret them). Please let me know if this clarifies?
